test/fail_compilation/retscope.d

```diff
@@ -349,3 +349,37 @@ int* bar11(scope int** x) @safe
 }
 
 int* foo11(int* x) @safe { return x; }
+
+/******************************************/
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/retscope.d(366): Error: escaping reference to local variable f
+fail_compilation/retscope.d(377): Error: scope variable f may not be returned
+---
+*/
+
+int* escape12() @safe
+{
+    Foo12 f;
+    return f.foo;
+}
+
+struct Foo12
+{
+    int* foo() return @safe;
+}
```

Comment:
This is not the test case I provided. The original test case was

```D
int* escape () @safe
{
    int i;
    Foo f;
    f.v = &i;
    return f.foo;
}

struct Foo
{
    int* v;
    int* foo () @safe { return this.v; }
}
```

And still compiles with this P.R.